### PR TITLE
[Hexagon]Pull and build specific LLVM sha for hexagon docker CI

### PR DESCRIPTION
**Pull request by joshherr-quic**
https://github.com/apache/tvm/pull/12873 needs an updated LLVM version, as noted in https://github.com/apache/tvm/issues/12966. This PR pulls the most recent llvm patch off of main which should work with the latest tvm.

### CODE REVIEW

**Comment by TejashShah** [comment]
cc @mehrdadh

File: docker/install/ubuntu_install_hexagon.sh [hunk 1]
```diff
@@ -21,9 +21,26 @@ set -o pipefail
 
 # Install LLVM/clang
 CLANG_LLVM_HOME=/opt/clang-llvm
-CLANG_LLVM_VERSION=14.0.0
-CLANG_LLVM_FILENAME=clang_llvm.tar.xz
-wget -q https://github.com/llvm/llvm-project/releases/download/llvmorg-${CLANG_LLVM_VERSION}/clang+llvm-${CLANG_LLVM_VERSION}-x86_64-linux-gnu-ubuntu-18.04.tar.xz -O ${CLANG_LLVM_FILENAME}
-mkdir ${CLANG_LLVM_HOME}
-tar -xvf ${CLANG_LLVM_FILENAME} -C ${CLANG_LLVM_HOME} --strip-components=1
-rm ${CLANG_LLVM_FILENAME}
+HEXAGON_LLVM_SHA=361a27c155ec8b222e3318488a208c0eb39624c8
```
Comment (mehrdadh) [inline docker/install/ubuntu_install_hexagon.sh:24]: Please rename this to LLVM_SHA

File: docker/install/ubuntu_install_hexagon.sh [hunk 2]
```diff
@@ -21,9 +21,26 @@ set -o pipefail
 
 # Install LLVM/clang
 CLANG_LLVM_HOME=/opt/clang-llvm
```
Comment (mehrdadh) [inline docker/install/ubuntu_install_hexagon.sh:23]: `CLANG_LLVM_HOME` is used in hexagon TVM build. Can you change this script to install LLVM in `CLANG_LLVM_HOME` path

**Review by mehrdadh — CHANGES_REQUESTED** [review]
Thanks for this PR. Please address my comments

**Review by mehrdadh — APPROVED** [review]
LGTM, thanks!
Next step is to wait for the docker image to be build and uploaded to the tlcpackstaging. Then you need to send another PR which updated the hexagon docker image tag. That's when it actually uses the docker and run tests inside the new docker

**Pull request merged**